### PR TITLE
[SPARK-33126][SQL] Simplify offset window function(Remove direction field)

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The current `Lead`/`Lag` extends `OffsetWindowFunction`. `OffsetWindowFunction` contains field `direction` and use `direction` to calculates the `boundary`.

We can use single literal expression unify the two properties.
For example:
3 means `direction` is Asc and `boundary` is 3.
-3 means `direction` is Desc and `boundary` is -3.

### Why are the changes needed?
Improve the current implement of `Lead`/`Lag`.

### Does this PR introduce _any_ user-facing change?
 'No'.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala [hunk 1]
```diff
@@ -348,16 +348,12 @@ abstract class OffsetWindowFunction
 
   /**
    * (Foldable) expression that contains the number of rows between the current row and the row
-   * where the input expression is evaluated.
+   * where the input expression is evaluated. If `offset` is a positive integer, it means that
+   * the direction of the `offset` is from front to back. If it is a negative integer, the direction
+   * of the `offset` is from back to front.
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala:353]: what if the offset is 0?

**Review by cloud-fan — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala]: `If it is zero, it means that the offset is ignored and use current row.`

File: sql/core/src/test/resources/sql-tests/results/postgreSQL/window_part1.sql.out [hunk 2]
```diff
@@ -202,7 +202,7 @@ struct<ntile(3) OVER (ORDER BY ten ASC NULLS FIRST, four ASC NULLS FIRST ROWS BE
 -- !query
 SELECT lag(ten) OVER (PARTITION BY four ORDER BY ten), ten, four FROM tenk1 WHERE unique2 < 10
 -- !query schema
-struct<lag(ten, 1, NULL) OVER (PARTITION BY four ORDER BY ten ASC NULLS FIRST ROWS BETWEEN -1 FOLLOWING AND -1 FOLLOWING):int,ten:int,four:int>
+struct<lag(ten, -1, NULL) OVER (PARTITION BY four ORDER BY ten ASC NULLS FIRST ROWS BETWEEN -1 FOLLOWING AND -1 FOLLOWING):int,ten:int,four:int>
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/results/postgreSQL/window_part1.sql.out:205]: The user query is `lag(ten)`. Can we override `stringArgs` in `OffsetWindowFunction` to only include `input`?

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/results/postgreSQL/window_part1.sql.out:205]: I don't remember the details, please check the code and find a way to change the string format here.

Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/results/postgreSQL/window_part1.sql.out:205]: I got it.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
GA passed, merging to master, thanks!

**Comment by beliefer** [comment]
@cloud-fan Thanks for your help!

**Pull request closed without merging**